### PR TITLE
DOC: Added math notation in examples in ltisys.py

### PR DESCRIPTION
**Pull request by nosseb**
Added `.. math::` notation to the transfer function in a bunch of examples, as it makes online documentation easier to read.

**This pull request does not cover State Space notation** as I'm unfamiliar with it and current doctring don't provide any baseline.

Also, I don't know how to build the html documentation, but it's basic TeX ~~so shouldn't cause any issue.~~
edit: Well it did cause issue because TeX uses backslashes. Had to set doc-strings to raw format.

### CODE REVIEW

**Comment by rgommers** [comment]
Thanks @nosseb. We try to use LaTeX sparingly to keep things readable in an editor or terminal, but it seems to be appropriate here. Fractions are easier to read rendered.

The doc build timed out due to an unrelated issue, I've rebased on master to get rid of that.

**Comment by rgommers** [comment]
CI doc build worked, other failures are unrelated. @nosseb do you want to add more changes, or is this ready? If the latter, can you move the PR out of "draft"?

File: scipy/signal/ltisys.py [hunk 1]
```diff
@@ -358,6 +362,8 @@ class dlti(LinearTimeInvariant):
     dt: 0.1
     )
 
+    .. math:: H(s) = \frac{5(z - 1)(z - 2)}{(z - 3)(z - 4)}
```
Comment (ilayn) [inline scipy/signal/ltisys.py:365]: This should be `H(z)`

File: scipy/signal/ltisys.py [hunk 2]
```diff
@@ -908,9 +918,11 @@ class ZerosPolesGain(LinearTimeInvariant):
 
     Examples
     --------
-    >>> from scipy import signal
+    Construct the transfer function:
 
-    Transfer function: H(s) = 5(s - 1)(s - 2) / (s - 3)(s - 4)
+    ..math:: H(s) = \frac{5(s - 1)(s - 2)}{(s - 3)(s - 4)}
```
Comment (ilayn) [inline scipy/signal/ltisys.py:923]: These won't render without indentation

Comment (ilayn) [reply scipy/signal/ltisys.py:923]: Also they can be inline no need for extra lines

File: scipy/signal/ltisys.py [hunk 3]
```diff
@@ -920,7 +932,9 @@ class ZerosPolesGain(LinearTimeInvariant):
     dt: None
     )
 
-    Transfer function: H(z) = 5(z - 1)(z - 2) / (z - 3)(z - 4)
+    Construct the transfer function with a sampling time of 0.1 seconds:
+
+    ..math:: H(s) = \frac{5(z - 1)(z - 2)}{(z - 3)(z - 4)}
```
Comment (ilayn) [inline scipy/signal/ltisys.py:937]: Same as above, inline is fine

File: scipy/signal/ltisys.py [hunk 4]
```diff
@@ -1101,9 +1115,11 @@ class ZerosPolesGainContinuous(ZerosPolesGain, lti):
 
     Examples
     --------
-    >>> from scipy import signal
+    Construct the transfer function:
 
-    Transfer function: H(s) = 5(s - 1)(s - 2) / (s - 3)(s - 4)
+    .. math:: H(s) = \frac{5(s - 1)(s - 2)}{(s - 3)(s - 4)}
```
Comment (ilayn) [inline scipy/signal/ltisys.py:1120]: Same issue as above

File: scipy/signal/ltisys.py [hunk 5]
```diff
@@ -1173,9 +1189,11 @@ class ZerosPolesGainDiscrete(ZerosPolesGain, dlti):
...
```
Comment (ilayn) [inline scipy/signal/ltisys.py:1194]: Same issue

File: scipy/signal/ltisys.py [hunk 6]
```diff
@@ -1185,7 +1203,9 @@ class ZerosPolesGainDiscrete(ZerosPolesGain, dlti):
     dt: None
     )
 
-    Transfer function: H(z) = 5(z - 1)(z - 2) / (z - 3)(z - 4)
+    Construct the transfer function with a sampling time of 0.1 seconds:
+
+    .. math:: H(s) = \frac{5(z - 1)(z - 2)}{(z - 3)(z - 4)}
```
Comment (ilayn) [inline scipy/signal/ltisys.py:1208]: Same issue

File: scipy/signal/ltisys.py [hunk 7]
```diff
@@ -2600,7 +2620,9 @@ def freqresp(system, w=None, n=10000):
     >>> from scipy import signal
     >>> import matplotlib.pyplot as plt
 
-    Transfer function: H(s) = 5 / (s-1)^3
+    Transfer function:
+    
+    .. math:: H(s) = \frac{5}{(s-1)^3}
```
Comment (ilayn) [inline scipy/signal/ltisys.py:2625]: Same issue

File: scipy/signal/ltisys.py [hunk 8]
```diff
@@ -3720,7 +3742,9 @@ def dfreqresp(system, w=None, n=10000, whole=False):
     >>> from scipy import signal
     >>> import matplotlib.pyplot as plt
 
-    Transfer function: H(z) = 1 / (z^2 + 2z + 3)
+    Transfer function:
+
+    .. math:: H(z) = \frac{1}{z^2 + 2z + 3}
```
Comment (ilayn) [inline scipy/signal/ltisys.py:3747]: Same issue and missing the dt info

File: scipy/signal/ltisys.py [hunk 9]
```diff
@@ -3814,7 +3838,9 @@ def dbode(system, w=None, n=100):
     >>> from scipy import signal
     >>> import matplotlib.pyplot as plt
 
-    Transfer function: H(z) = 1 / (z^2 + 2z + 3)
+    Transfer function:
+    
+    .. math:: H(z) = \frac{1}{z^2 + 2z + 3}
```
Comment (ilayn) [inline scipy/signal/ltisys.py:3843]: Same issue and missing the dt info

**Review by ilayn — CHANGES_REQUESTED** [review]
I've placed some minor remarks, you can see the results in the CI artifact https://26703-1460385-gh.circle-artifacts.com/0/html-scipyorg/generated/scipy.signal.ZerosPolesGain.html#scipy.signal.ZerosPolesGain

**Comment by nosseb** [comment]
Thanks @rgommers for explaining the timing-out build. Also I had been unsuccessful in setting-up build pipeline on my machine and thus kinda dropped the whole thing for a moment.

I hope my last few commits should satisfy @ilayn . I took the liberty to apply the inline equation trick to older bits of documentation as well.

I first wanted to add documentation for state space notation as well. As I have to ask a teacher for some help on that, this will have to be implemented in a subsequent pull request.

**Review by ilayn — APPROVED** [review]

**Review by rgommers — APPROVED** [review]
LGTM too now, thanks @nosseb. And thanks @ilayn for reviewing.

**Pull request merged**